### PR TITLE
changed interfaces and CsvProvider implementation

### PR DESCRIPTION
**Pull request by wpeisert**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no

### CODE REVIEW

**Comment by wpeisert** [comment]
I've changed interfaces only.

**Comment by dpfaffenbauer** [comment]
it looks fine to me, but I have to test it first

File: src/ImportDefinitionsBundle/Provider/ImportDataSet.php [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+ * For the full copyright and license information, please view the LICENSE.md and gpl-3.0.txt
+ * files that are distributed with this source code.
+ *
+ * @copyright  Copyright (c) 2016-2018 w-vision AG (https://www.w-vision.ch)
+ * @license    https://github.com/w-vision/ImportDefinitions/blob/master/gpl-3.0.txt GNU General Public License version 3 (GPLv3)
+ */
+
+namespace ImportDefinitionsBundle\Provider;
+
+class ImportDataSet implements ImportDataSetInterface
```
Comment (dpfaffenbauer) [inline src/ImportDefinitionsBundle/Provider/ImportDataSet.php]: do we actually need this class? It does the same as the actual Iterator you would have to pass.

Comment (wpeisert) [reply src/ImportDefinitionsBundle/Provider/ImportDataSet.php]: It implements count(). Passing iterator would result in "method count() not implemented" error.

Comment (dpfaffenbauer) [reply src/ImportDefinitionsBundle/Provider/ImportDataSet.php]: that is true :)

**Comment by dpfaffenbauer** [comment]
thanks a lot

**Pull request merged**